### PR TITLE
Add Helm chart

### PR DESCRIPTION
**Pull request by jonkerj**
As discussed in #3374, we'd like to submit our Helm chart for inclusion at Zipkin. It may need some CD or other stuff, all feedback is welcome!

### CODE REVIEW

**Comment by jcchavezs** [comment]
Test is failing because of lack of license in YAML files. Not sure this is truly needed:

image

Shall we @basvanbeek @abesto?

**Comment by basvanbeek** [comment]
yes we should have the license headers in.

**Comment by jonkerj** [comment]
No problem, I've added them. Can someone trigger a re-test?

**Comment by jonkerj** [comment]
Hm, the headers are there, albeit probabky in a different format your test expects. I've used Helm block comments (`{{- /* ... */ }}`) where possible, as the license applies to the templates and not to the generated manifests. I suspect the test does not like that. Although `Chart.yaml` and `values.yaml` contain simple YAML-comments (as they are not templated) and also make the test angry. Does the test work at all for YAML? Anything I can do about this?

**Comment by jonkerj** [comment]
Did some digging in [license-maven-plugin](https://github.com/mojohaus/license-maven-plugin), and [it seems to me](https://github.com/mojohaus/license-maven-plugin/tree/master/src/main/java/org/codehaus/mojo/license/header/transformer) that it is limited in types of files / comment styles supported. I cannot seem to find YAML or Helm-style comments in there. 

One could (in theory) submit YAML/Helm support into that plugin, and bump the license-mavin-plugin version in this repo afterwards, but I am not eager to jump on that boat.

**Comment by jcchavezs** [comment]
I am inclined to say we should ignore that folder for now but I'd like to
hear from @basvanbeek and @llinder. Also would be cool to see @abesto
around here.

**Comment by abesto** [comment]
Who me? Hi! :D
TBH I don't think I should have a word on setting the requirements for this - that is, whether it's OK to not have license headers in these files. I know this was extremely important under ASF, I'm not sure how much we care about it now. Personally I wouldn't worry about ignoring all YAML files.

On the technical front, AFAICT YAML supports line comments starting with `#`, which the Maven plugin also supports with the `properties` comment style. So possibly it's a matter of submitting a PR to the plugin that adds `.yaml` to the default extension list? Though note, that transformer also understands hashbangs, which are invalid in YAML, but whatever. Better yet, https://www.mojohaus.org/license-maven-plugin/check-file-header-mojo.html#extraExtensions looks like it's possible to add an extra extension in configuration. So maybe that'd fly?

**Comment by jcchavezs** [comment]
@jonkerj could we avoid that folder to move on? cc @basvanbeek

**Comment by jonkerj** [comment]
> @jonkerj could we avoid that folder to move on? cc @basvanbeek

Sure, but I wouldn't know how. I am not familiar with the build system (it's maven, right?) and got lost trying to find this out. If someone can point me to other stuff getting ignored by this check, I'll see that I replicate that to my folder.

**Comment by jonkerj** [comment]
Bump

**Comment by mrfelek** [comment]
@jonkerj
ingress.yaml:
`{{- $svcPort := .Values.service.port -}}`   should be `{{- $svcPort := .Values.service.zipkin.port -}}`
values.yaml:
`service:
  tracing:
    type: ClusterIP
    port: 80
  zipkin:
    type: ClusterIP
    port: 9411
`
Anyway... good work.. Thank you!

**Comment by jonkerj** [comment]
@mrfelek : thanks for the sharp eye, don't know how we could have missed that. Still looking for hints on how to exclude the chart subdir from the license header check. Can anyone point me towards the right direction?

**Comment by jcchavezs** [comment]
@jonkerj are you on twitter? I'd like to have a word with you.

**Comment by jonkerj** [comment]
I'm sorry, I'm not on Twitter. I'm on several slacks (Talos, CNCF, Cilium, Kubernetes), though

**Comment by jonkerj** [comment]
Don't want to sound impatient, but I've got the feeling this PR and the conversation around it is stalling a bit. Don't get me wrong: we really love to help in getting this chart included, and believe it would boost the adaption of Zipkin in a Kubernetes environment. 

If this has no priority from the main devs: totally fine and we'd respect that, and we will find a different approach, but please be clear about this.

**Comment by jcchavezs** [comment]
Let's merge this I will address the headers in a next PR.

File: chart/zipkin-server/values.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+securityContext:
+  # capabilities:
+  #   drop:
+  #   - ALL
+  readOnlyRootFilesystem: true
+  runAsNonRoot: true
+  runAsUser: 1000
+
+service:
+  tracing:
```
Comment (jcchavezs) [inline chart/zipkin-server/values.yaml:51]: Curious, why do we need this? What is it exposed under 80?

File: chart/zipkin-server/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      serviceAccountName: {{ include "zipkin.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          env:
+            - name: STORAGE_METHOD
```
Comment (jcchavezs) [inline chart/zipkin-server/templates/deployment.yaml:50]: Just curious, what is this exactly?

Comment (jonkerj) [reply chart/zipkin-server/templates/deployment.yaml:50]: Not really sure what you mean, but let me take you through it line by line:
47: inserts the security context settings
48: sets the image to be used, defaulting to AppVersion from Chart.yaml
49/50: half of the code setting the storage backend to `mem`

**Pull request merged**